scipy/stats/tests/test_morestats.py

```diff
@@ -979,6 +979,23 @@ def test_mood_bad_arg(self):
         # less than 3
         assert_raises(ValueError, stats.mood, [1], [])
 
+    def test_mood_alternative(self):
+        np.random.seed(0)
+        x = stats.norm.rvs(scale=0.75, size=100)
+        y = stats.norm.rvs(scale=1.25, size=100)
+
+        stat1, p1 = stats.mood(x, y, alternative='two-sided')
+        stat2, p2 = stats.mood(x, y, alternative='less')
+        stat3, p3 = stats.mood(x, y, alternative='greater')
+
+        assert stat1 == stat2 == stat3
+        assert_allclose(p1, 0, atol=1e-7)
+        assert_allclose(p2, p1/2)
+        assert_allclose(p3, 1 - p1/2)
+
```

Comment:
@tirthasheshpatel I will add a test against R to check that the convention used here is the same. But the first test was carefully designed to be intuitive (easy to verify) and I think it achieves that. 

One can tell by inspection that the scale of distribution underlying the first sample is less than the scale of the distribution underlying the second sample because the `scale` is given explicitly; it does not need to be inferred from a list of numbers. 

It is also easy to see that any detailed requirements of the test are satisfied because the samples come from normal distributions with the same location (see notes). This is difficult to infer from a list of numbers.

So immediately one can tell that the p-value of a two-sided comparison should be _low_. 
Indeed, the two-sided p-value is nearly zero (which is less than 0.05 - any other chosen significance level preferred by the reader).

The test is also strong. Rather than just testing that the p-value of the `less` alternative is small (< 0.05), it tests that it is half the two-sided p-value, which we know it should by definition Similarly, we can see that the p-value of the `greater` alternative is large (0.95), and, more than that, it is the complement of the `less` p-value.

So I don't think I would change the style of the test. 
When I suggest in gh-13549 that we add more intuitive tests, these are the qualities I was looking for. 
Do you not agree that the current tests are intuitive and strong?

Reply:
> When I suggest in gh-13549 that we add more intuitive tests, these are the qualities I was looking for.

Ah, I get it. Thanks for the explanation!

> Do you not agree that the current tests are intuitive and strong?

No no, I do agree that the current tests are strong enough. So, no worries about adding this suggested test!

> I will add a test against R to check that the convention used here is the same

OK, thanks!